applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -144,54 +144,78 @@ static void le_audio_event_publish(enum le_audio_evt_type event, struct bt_conn
  */
 static int headset_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
 {
-	uint32_t pres_dly_min = headsets[index].sink_ep->qos_pref.pd_min;
-	uint32_t pres_dly_max = headsets[index].sink_ep->qos_pref.pd_max;
-	uint32_t pref_dly_min = headsets[index].sink_ep->qos_pref.pref_pd_min;
-	uint32_t pref_dly_max = headsets[index].sink_ep->qos_pref.pref_pd_max;
+	uint32_t pd_min = headsets[index].sink_ep->qos_pref.pd_min;
+	uint32_t pd_max = headsets[index].sink_ep->qos_pref.pd_max;
+	uint32_t pref_pd_min = headsets[index].sink_ep->qos_pref.pref_pd_min;
+	uint32_t pref_pd_max = headsets[index].sink_ep->qos_pref.pref_pd_max;
 
 	LOG_DBG("Index: %d, Pref min: %d, pref max: %d, pres_min: %d, pres_max: %d", index,
-		pref_dly_min, pref_dly_max, pres_dly_min, pres_dly_max);
+		headsets[index].sink_ep->qos_pref.pref_pd_min,
+		headsets[index].sink_ep->qos_pref.pref_pd_max,
+		headsets[index].sink_ep->qos_pref.pd_min, headsets[index].sink_ep->qos_pref.pd_max);
+
+	*pres_dly_us = 0;
 
 	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
-		if (headsets[i].sink_ep != NULL) {
-			pres_dly_min = MAX(pres_dly_min, headsets[i].sink_ep->qos_pref.pd_min);
-			pres_dly_max = MIN(pres_dly_max, headsets[i].sink_ep->qos_pref.pd_max);
-			pref_dly_min = MAX(pref_dly_min, headsets[i].sink_ep->qos_pref.pref_pd_min);
-			pref_dly_max = MIN(pref_dly_max, headsets[i].sink_ep->qos_pref.pref_pd_max);
+		if (le_audio_ep_qos_configured(headsets[i].sink_ep)) {
+			LOG_DBG("i: %d, Pref min: %d, pref max: %d, pres_min: %d, pres_max: %d", i,
+				headsets[i].sink_ep->qos_pref.pref_pd_min,
+				headsets[i].sink_ep->qos_pref.pref_pd_max,
+				headsets[i].sink_ep->qos_pref.pd_min,
+				headsets[i].sink_ep->qos_pref.pd_max);
+
+			pd_min = MAX(pd_min, headsets[i].sink_ep->qos_pref.pd_min);
+			pref_pd_min = MAX(pref_pd_min, headsets[i].sink_ep->qos_pref.pref_pd_min);
+
+			if (headsets[i].sink_ep->qos_pref.pd_max) {
+				pd_max = MIN(pd_max, headsets[i].sink_ep->qos_pref.pd_max);
+			}
+
+			if (headsets[i].sink_ep->qos_pref.pref_pd_max) {
+				pref_pd_max =
+					MIN(pref_pd_max, headsets[i].sink_ep->qos_pref.pref_pd_max);
+			}
 		}
 	}
 
 	if (IS_ENABLED(CONFIG_BT_AUDIO_PRES_DELAY_SRCH_MIN)) {
-		*pres_dly_us = pres_dly_min;
+		*pres_dly_us = pd_min;
 
 		return 0;
 	}
 
 	if (IS_ENABLED(CONFIG_BT_AUDIO_PRES_DELAY_SRCH_MAX)) {
-		*pres_dly_us = pres_dly_max;
+		*pres_dly_us = pd_max;
 
 		return 0;
 	}
 
 	if (IS_ENABLED(CONFIG_BT_AUDIO_PRES_DELAY_SRCH_PREF_MIN)) {
-		/* Preferred min is 0, so we set min supported */
-		if (pref_dly_min == 0) {
-			*pres_dly_us = pres_dly_min;
+		if (pref_pd_min == 0) {
+			*pres_dly_us = pd_min;
+		} else if (pref_pd_min <= pd_max) {
+			*pres_dly_us = pref_pd_min;
 		} else {
```

Comment:
```suggestion
		if (pref_pd_min == 0) {
			*pres_dly_us = pd_min;
		} else if (pref_pd_min = pd_max) {
			*pres_dly_us = pref_pd_min;
		} else if (pref_pd_min < pd_max) {
			*pres_dly_us = pref_pd_min;
			LOG_WRN("....");
		} else {
```

Reply:
And the same below.

Reply:
Not sure I follow the above, why the warning?